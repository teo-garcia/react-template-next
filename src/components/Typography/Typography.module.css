.regular {
  font-weight: var(--font-weight-primary);
}

.bold {
  font-weight: var(--font-weight-secondary);
}

.background {
  color: var(--color-background);
}

.text {
  color: var(--color-text);
}

.primary {
  color: var(--color-primary);
}

.secondary {
  color: var(--color-secondary);
}
