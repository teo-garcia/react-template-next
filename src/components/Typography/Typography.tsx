import styles from "./Typography.module.css";
import { TypographyProps } from "../../types/client";
import classNames from "classnames";

function Typography(props: TypographyProps) {
  const {
    tagName = "p",
    className,
    weight = "regular",
    color = "text",
    ...rest
  } = props;
  const Tag = tagName as keyof JSX.IntrinsicElements;
  return (
    <Tag
      className={classNames(
        "typography",
        styles[weight],
        styles[color],
        className
      )}
      {...rest}
    />
  );
}

export { Typography };
