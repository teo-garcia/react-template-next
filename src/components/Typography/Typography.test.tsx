import { render, screen } from '@testing-library/react';
import { Typography } from './Typography';

describe('<Typography /> tests', function () {
  test('Should match with the component snapshot', function () {
    const { asFragment } = render(<Typography>Hello World</Typography>);
    expect(asFragment()).toMatchSnapshot();
  });

  test('Should render the provided children prop', function () {
    render(<Typography>Hello World</Typography>);
    expect(screen.getByText(/Hello World/i)).toBeInTheDocument();
  });

  test('Should render the provided tagName prop', function () {
    render(<Typography tagName="h1">Hello World</Typography>);
    expect(screen.getByText(/Hello World/i).tagName).toBe('H1');
  });
});
