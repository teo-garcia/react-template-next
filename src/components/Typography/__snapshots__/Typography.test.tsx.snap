// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Typography /> tests Should match with the component snapshot 1`] = `
<DocumentFragment>
  <p
    class="typography typography--regular typography--text"
  >
    Hello World
  </p>
</DocumentFragment>
`;
